ChatController.m

```diff
@@ -77,14 +86,31 @@ - (void)sendRequestToOpenAIWithMessage:(NSString *)message {
 
     // Check if proxyValue is empty or nil before proceeding
     if (proxyValue == nil || [proxyValue isEqualToString:@""]) {
-		[proxyServer setStringValue:@"Please enter a value!"];
-        NSLog(@"Proxy server value is missing.");
-        return;
+        proxyValue = nil;
+        NSArray *addresses = [[NSHost currentHost] addresses];
+
+        // Get local IP address and tack on 8080 by default
+        NSEnumerator *e = [addresses objectEnumerator];
+        NSString *anAddress;
+        while (anAddress = [e nextObject]) {
+            if (![anAddress hasPrefix:@"127"] && [[anAddress componentsSeparatedByString:@"."] count] == 4) {
+                proxyValue = [NSString stringWithFormat:@"%@:8080", anAddress];
+                break;
+            }
+        }
```

Comment:
I like this, very readable

Reply:
Thanks! I borrowed the idea from [this Stackoverflow](https://stackoverflow.com/a/3268554/1170723) and rewrote it to work with 10.4. I found it _very_ strange that neither `127.0.01` nor `localhost` worked, it would fail to connect each time... I couldn't figure out why, so I instead just looked for how to get the local IP address.